src/teehr/visualization/dataframe_accessor.py

```diff
@@ -0,0 +1,122 @@
+"""Provides the teehr accessor extending pandas DataFrames"""
+import itertools
+from math import pi
+import random
+import pandas as pd
+
+from bokeh.plotting import figure, save, show
+from bokeh.palettes import Turbo256
+
+@pd.api.extensions.register_dataframe_accessor("teehr")
+class TEEHRDataFrameAccessor:
+    """Extends pandas DataFrame objects.
+
+    Notes
+    -----
+    This class contains example methods for summarizing and plotting metrics
+    as well as timeseries. This requires more validation in each method to
+    ensure the DataFrame has the appropriate data.
+
+    Methods operating on metrics data should start with 'metrics_' and methods
+    operating on timeseries data should start with 'timeseries_'.
+    """
+
+    def __init__(self, pandas_obj):
+        """Initialize the class."""
+        self._df = pandas_obj
+        self._validate(pandas_obj)
+
+    @staticmethod
+    def _validate(obj):
```

Comment:
I think the main plotting methods should have some validation checking if the `df` contains the fields the user is asking for, and returning a message if not